doc/dev/sharp_bits.rst

```diff
@@ -509,133 +509,14 @@ The optimization now takes 574ms ± 43.1ms to complete when using 200 steps.
-a qjit-compiled function, as long as the return shape and type is known:
-
-.. code-block:: python
-
-    @qjit
-    def fn(x):
-
-        A = jnp.sin(x) * jnp.array([0.23, 0.2], [0.43, -0.54.])
-
-        @catalyst.pure_callback
-        @jax.jit # since the callback function is pure JAX, we can jit it
-        def callback_fn(A) -> jax.ShapeDtypeStruct(A.shape, A.dtype):
-            # here we call non-Catalyst compatible code
-            return jax.scipy.linalg.expm(A)
-
-        return jnp.cos(callback_fn(A))
-
->>> fn(0.654)
-array([[0.39385058, 0.99369752],
-       [0.97097762, 0.74283208]])
-
-Catalyst provides several callback functions:
-
-- :func:`~.pure_callback` supports callbacks of **pure** functions. That is, functions with no
-  side-effects that accept parameters and return values. However, the return type and shape of the
-  function must be known in advance, and is provided as a type signature.
-
-- :func:`~.accelerate` is similar to :func:`~.pure_callback` above, but is designed to
-  work only with functions that are ``jax.jit`` compatible. As a result of this restriction,
-  return types do not have to be provided upfront, and support is provided for executing
-  these callbacks directly on classical accelerators such as GPUs and TPUs.
-
-- :func:`~.debug.callback` supports callbacks of functions with **no** return values. This makes it
-  an easy entry point for debugging, for example via printing or logging at runtime.
-
-Note that to use :func:`~.pure_callback` within functions that are being differentiated,
-a custom VJP rule **must** be defined so that the Catalyst compiler knows how to
-differentiate the callback. This can be done via the ``pure_callback.fwd`` and
-``pure_callback.bwd`` methods. See the :func:`~.pure_callback` documentation for
-more details.
-
-JAX integration
----------------
-
-Compiled functions remain JAX compatible, and you can call JAX transformations
-on them, such as ``jax.grad`` and ``jax.vmap``. You can even call ``jax.jit``
-on functions that call qjit-compiled functions:
-
->>> dev = qml.device("lightning.qubit", wires=2)
->>> @qjit
-... @qml.qnode(dev)
-... def circuit(x):
-...     qml.RX(x, wires=0)
-...     return qml.expval(qml.PauliZ(0))
->>> @jax.jit
-... def workflow(y):
-...     return jax.grad(circuit)(jnp.sin(y))
->>> workflow(0.6)
-Array(-0.53511382, dtype=float64, weak_type=True)
-
-However, a ``jax.jit`` function calling a ``qjit`` function will always result
-in a callback to Python, so will be slower than if the function was purely compiled
-using ``jax.jit`` or ``qjit``.
-
-If you want to compile some functionality that is not currently Catalyst
-compatible, or you want to make use of JAX-supported hardware such as TPUs
-for classical processing, mixing ``jax.jit`` and ``qjit`` will allow this.
-However, if possible, try to always use ``qjit`` to compile your entire
-workflow.
+.. note::
 
-Internal QJIT transformations
------------------------------
```

Comment:
I've kept this section in the sharp bits guide, since it is actually a sharp bit.